frontend/src/actions.js

```diff
@@ -17,4 +17,111 @@ const logout = () =>
     credentials: 'same-origin',
   }).then(res => res.json());
 
-export default { getUser, logout };
+const slug = fullPath =>
+  fullPath
+    .toLowerCase()
+    .replace(/[^A-Za-z0-9-| ]/g, '')
+    .replace(/\s/g, '-');
+
+const DEFAULT_DISPLAY_SUBTOPIC = 'Uncategorized Articles';
```

Comment:
do we need this? An article without a sub-topic should just be displayed as a leaf node under the topic, I don't want to see it listed under "uncategorised Articles"

Reply:
it is possible to make, first thought it will break existing order, i'll look but warn it can take time

Reply:
not sure what 'existing order' you're referring to? Do you mean e.g.
testing|first article
testing|sub-topic|another article
testing|sub-topic2|yet another article

Reply:
yes i mean exactly that. every case has now the same pattern. but as i said earlier, i ll check what i can do.